File: ED/atletas.c
#include <stdio.h>
#include <string.h>

struct Atleta {
    char nome[100];
    int id, n_ouro, n_prata, n_bronze;
};

int compare_conqui(int atleta1, int atleta2, struct Atleta vatletas[]){
    int pos1 = 0, pos2 = 0;
    for (int c = 0; c < 100; c++){
        if (vatletas[c].id == atleta1){
            pos1 = c;
            break;
        }
    }

    for (int c = 0; c < 100; c++){
        if (vatletas[c].id == atleta2){
            pos2 = c;
            break;
        }
    }

    if (vatletas[pos1].n_ouro > vatletas[pos2].n_ouro){
        return atleta1;
    } else if(vatletas[pos1].n_ouro < vatletas[pos2].n_ouro){
        return atleta2;
    } else if(vatletas[pos1].n_prata > vatletas[pos2].n_prata){
        return atleta1;
    }else if(vatletas[pos1].n_prata < vatletas[pos2].n_prata){
        return atleta2;
    } else if(vatletas[pos1].n_bronze > vatletas[pos2].n_bronze){
        return atleta1;
    }else if(vatletas[pos1].n_bronze < vatletas[pos2].n_bronze){
        return atleta2;
    }else{
        return 0;
    }
}


int main(void){
    int consultanum = 0, atletanum = 0;
    struct Atleta vatletas[100];

    scanf("%d", &atletanum);
    for (int c = 0; c < atletanum; c++){
        scanf("%d %s", &vatletas[c].id , vatletas[c].nome);
        scanf("%d %d %d", &vatletas[c].n_ouro, &vatletas[c].n_prata, &vatletas[c].n_bronze);

    }

    scanf("%d", &consultanum);

    int a1, a2, resultado;
    for (int c = 0; c < consultanum; c++){
        scanf("%d %d", &a1, &a2);
        resultado = compare_conqui(a1, a2, vatletas);
        (resultado == 0) ? printf("empate\n") : printf("%d\n", resultado);
    }

    return 0;
}

File: ED/atletas.py
class Atleta:
    def __init__(self, id, n_ouro, n_prata, n_bronze, nome):
        self.id = id
        self.n_ouro = n_ouro
        self.n_prata = n_prata
        self.n_bronze = n_bronze
        self.nome = nome

    def comparar_conqui(self, atleta2):
        if self.n_ouro > atleta2.n_ouro:
            return self.id
        elif self.n_ouro < atleta2.n_ouro:
            return atleta2.id
        elif self.n_prata > atleta2.n_prata:
            return self.id
        elif self.n_prata < atleta2.n_prata:
            return atleta2.id
        elif self.n_bronze > atleta2.n_bronze:
            return self.id
        elif self.n_bronze < atleta2.n_bronze:
            return atleta2.id
        
        return "empate"
    
    #Método para verificar se o id corresponde ao atleta
    def busca_id(self, id):
        if self.id == id:
            return True
        
        return False


atletanum = int(input())

vatletas = []
for c in range(0, atletanum):
    atleta = input()
    atleta = atleta.split()
    medalha = input()
    medalha = medalha.split()

    #Coloca todas as informações organizadas na lista de Atletas
    vatletas.append(Atleta(int(atleta[0]), int(medalha[0]), int(medalha[1]), int(medalha[2]), atleta[1]))

consultanum = int(input())

#Procura dentro da lista de Atletas
for c in range(0, consultanum):
    compara = input()
    compara = compara.split()
    for d in vatletas:
        #Verifica se o id procurado é o mesmo do atleta
        if d.busca_id(int(compara[0])):
            #Procura se o segundo id é o mesmo do atleta
            for j in vatletas:
                if j.busca_id(int(compara[1])):
                    print(d.comparar_conqui(j))
